torch/nn/modules/module.py

```diff
@@ -802,8 +803,12 @@ def compute_should_use_set_data(tensor, tensor_applied):
             with torch.no_grad():
                 param_applied = fn(param)
             p_should_use_set_data = compute_should_use_set_data(param, param_applied)
+
+            # subclasses may have multiple child tensors so we need to use swap_tensors
+            p_should_use_swap_tensors = should_use_swap_tensors or is_traceable_wrapper_subclass(param_applied)
```

Comment:
btw sorry I just caught this but wanted to note that this only updates the condition to `swap_tensors` for parameters, but not their associated `.grad` field, if it exists.

`should_use_swap_tensors` is [used again below for gradients on L837](https://github.com/pytorch/pytorch/pull/122755/files#diff-4be56271f7bfe650e3521c81fd363da58f109cd23ee80d243156d2d6ccda6263L832)

might we want to also update the condition for gradients?

Reply:
thanks for catching that -- updated!